src/expr.rs

```diff
@@ -298,9 +308,57 @@ macro_rules! numeric (
 	($i:expr, $f:expr ) => (map_opt!($i,call!($f),EvalResult::as_numeric));
 );
 
+enum TypeCast {
+	SignedInt(usize),
+	UnsignedInt(usize),
+}
+
+impl EvalResult {
+	fn cast(self, ctype: TypeCast) -> Self {
+		use self::TypeCast::*;
+		let round = |bits| 2i64.pow((bits) as u32);
+		let unsigned_mask = |i, b| (i+round(b*8)) % round(b*8);
+		let signed_mask = |i, b| {
+			let rounded = unsigned_mask(i, b);
+			let half = round(8*b-1);
+			if rounded >= half {rounded - half * 2} else { rounded }
+		};
+		match self {
+			EvalResult::Float(f) => match ctype {
+				SignedInt(b) => EvalResult::Int(Wrapping(signed_mask(f as i64, b))),
+				UnsignedInt(b) => EvalResult::Int(Wrapping(unsigned_mask(f as i64, b))),
+			},
+			EvalResult::Int(Wrapping(i)) => match ctype {
+				SignedInt(b) => EvalResult::Int(Wrapping(signed_mask(i, b))),
+				UnsignedInt(b) => EvalResult::Int(Wrapping(unsigned_mask(i, b))),
+			},
+			_ => self
+		}
+	}
+}
+
+
 impl<'a> PRef<'a> {
+	method!(cast<PRef<'a>,&[Token],TypeCast,::Error>, mut self,
+		delimited!(p!("("), alt!(
+			do_parse!(k!("unsigned") >> k!("int") >> (TypeCast::UnsignedInt(size_of::<c_uint>()))) |
```

Comment:
This will choose the size based on the platform that bindgen runs on (generally the compilation host), that seems incorrect.

Reply:
Should we handle cross compilation? unsigned int may be different size on different architectures.

Reply:
Yes we should, but I don't have a good idea for *how*.